CHANGELOG.md

```diff
@@ -8,6 +8,7 @@
 * Authentication support for Cortex remote-write sink. Thanks, [oscil8](https://github.com/oscil8)!
 * Option to flush sinks on shutdown. Thanks, [csolidum](https://github.com/csolidum)!
 * `trace.StartTrace` and `trace.StartChildSpan` now scale better across multiple goroutines.  Thanks [bpowers](https://github.com/bpowers)
+* Ensure concurrent flushes finish in order.  This prevents metric drops when flush on shutdown is enabled.  Thanks, [akutta](https://github.com/akutta)!
```

Comment:
What about tracking outstanding Flushes (e.g. WaitGroup) and wait for that to hit 0 (or some timeout) on shutdown?

I'd think that'd have less requirement for testing because it would be producing a readably signal from Flush rather than modifying Flush (and would only do a behavior change on Shutdown)

Reply:
I've found some other interesting behaviours that were causing other issues so i'm going to close this PR for now.

Once I get a bulk of the issues resolved, i'll open a new PR.  I did end up switching to a WaitGroup since the mutex ended up causing odd Flush behaviour when not shutting down.  In some cases, upwards of 5 minutes before attempting the first flush.